src/shogun/kernel/CombinedKernel.h

```diff
@@ -124,7 +124,14 @@ class CCombinedKernel : public CKernel
 		 */
 		inline CKernel* get_kernel(int32_t idx)
 		{
-			return (CKernel*) kernel_array->get_element(idx);
+			if (idx < get_num_kernels())
+			{
+				return (CKernel*)kernel_array->get_element(idx);
+			}
+			else
+			{
+				return 0;
```

Comment:
Ah I see.
The old code does this horrible thing ;)
In that case, I think it is OK to do this here. (Though a patch that cleans this is always welcome)